### PR TITLE
Integrate Goals API

### PR DESCRIPTION
**Pull request by ConnorBoulais**
This PR integrates the Goals API with our Goals component. I refactored the GoalsScreen into three separate files, since they got fairly large and complex after adding the queries/mutations.

This PR also makes the changes noted from my code review in App.js.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,19 +3,19 @@ import Amplify, { Auth, Hub } from 'aws-amplify';
 import { ApolloProvider } from '@apollo/client';
 import { ApolloClient, InMemoryCache, HttpLink } from '@apollo/client';
 import {Route, Switch} from 'react-router-dom';
-import CourseScreen from './Components/CourseScreen.js';
```
Comment (Bart-Ilg) [inline src/App.js:6]: goood to clear unneeded imports

File: src/App.js [hunk 2]
```diff
@@ -36,14 +36,29 @@ Amplify.configure({
 function App() {
   const [accessToken, setAccessToken] = useState(null);
   const hist = useHistory();
+
+  /*
+   * This function will refresh the current users credentials,
+   * if their credentials are present. The Auth.currentSession()
+   * throws an error if the user is not signed in, but this is
```
Comment (Bart-Ilg) [inline src/App.js:43]: good descriptions

File: src/App.js [hunk 3]
```diff
@@ -36,14 +36,29 @@ Amplify.configure({
...
+   * not actual error to us, since this will happen anytime the user
+   * opens the app but is not signed in. That is why the catch block
+   * is empty in this function */
   const refreshCredentials = () => {
     return Auth.currentSession()
-    .then(session => {
-      setAccessToken(session.getAccessToken());
```
Comment (Bart-Ilg) [inline src/App.js:42]: small spacing change, surprised git thought it was bigger

File: src/App.js [hunk 4]
```diff
@@ -136,38 +161,33 @@ function App() {
       
       <div className="navbar">
         <p className="title">flipt.ED</p>
-        {accessToken == null? <li><a onClick={() => Auth.federatedSignIn()}>Sign In</a></li> :
```
Comment (Bart-Ilg) [inline src/App.js:139]: good spacing cleanup

File: src/Components/CourseScreen.css [hunk 5]
```diff
@@ -47,13 +47,13 @@
   background-color: #F21E1D;
 }
 
-form {
+/* form {
```
Comment (Bart-Ilg) [inline src/Components/CourseScreen.css:50]: uncommenting and commenting different options, obviously change for final build, but good for devlopment mid project

File: src/Components/Goal.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
+import React, { useState  } from 'react';
```
Comment (Bart-Ilg) [inline src/Components/Goal.js:1]: Component looks good, formatting is very clean

File: src/Components/GoalForm.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+  root: {
+    '& > *': {
+      margin: theme.spacing(1),
+      width: '25ch',
+    },
+  },
+}));
+
+function GoalForm() {
+  //used by the form as a new goal is built
```
Comment (Bart-Ilg) [inline src/Components/GoalForm.js:18]: good to add descriptions for functions, makes it easier for others to edit

File: src/Components/GoalList.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+  function UnstarredGoalList() {
+    return goals.filter(({category, favorited}) => {return (category === (searchCategory) || searchCategory === '') && !favorited;}).map((goal) => (
+      <Goal key={goal.id} goal={goal}/>
+    ));
+  }
+
+  
+
+  function GoalsDisplay() {
```
Comment (Bart-Ilg) [inline src/Components/GoalList.js:68]: good and clean code

File: src/Components/GoalsScreen.css [hunk 9]
```diff
@@ -14,19 +14,10 @@
   align-items:stretch;
 }
 
-.demoLabel {
```
Comment (Bart-Ilg) [inline src/Components/GoalsScreen.css:17]: css tweaks, good to simplify where possible

File: src/Components/Mission.css [hunk 10]
```diff
@@ -54,10 +62,12 @@
     width: 100%;
 }
 
-ul {
+.MissonOverview ul {
```
Comment (Bart-Ilg) [inline src/Components/Mission.css:65]: good to specify for improved precision

**Review by Bart-Ilg — APPROVED** [review]
overall clean and well written code. Good practices in spacing and organization. Good comments on many functions, when there isn't a comment the name is self explanatory and code is simple. Excellent work

File: src/Components/GoalForm.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
+import { useMutation } from '@apollo/client';
```
Comment (tddumont) [inline src/Components/GoalForm.js:1]: Glad to see the Goal Form factored out into its own component!

File: src/Components/GoalsScreen.js [hunk 12]
```diff
@@ -1,268 +0,0 @@
-import React, { useState  } from 'react';
```
Comment (tddumont) [inline src/Components/GoalsScreen.js:1]: Good idea to break this into multiple different components!

**Review by tddumont — APPROVED** [review]
This code is very well organized. It looks like you did a lot of excellent work!

**Pull request merged**